### PR TITLE
Document fsGroup setting in securityContext

### LINKED ISSUE #4781: Document fsGroup setting in securityContext
Part of https://github.com/elastic/cloud-on-k8s/issues/2791 but also relevant for custom Elastic Stack Docker images that don't use the root user (e.g. Iron Bank)

If run as root we set up file system permissions in an initContainer and things "just work". We should however document that the data directory has to be made writeable for the `elasticsearch` user when not running the container as root.

### PR DESCRIPTION
**Pull request by thbkrkr**
Resolves #4781.

### CODE REVIEW

File: docs/orchestrating-elastic-stack-applications/elasticsearch-specification.asciidoc [hunk 1]
```diff
@@ -34,6 +34,7 @@ Before you deploy and run ECK, take some time to look at the basic and advanced
 - <<{p}-prestop>>
 - <<{p}-autoscaling>>
 - <<{p}-jvm-heap-dumps>>
+- <<{p}-securiy-context>>
```
Comment (pebrc) [inline docs/orchestrating-elastic-stack-applications/elasticsearch-specification.asciidoc:37]:
```suggestion
- <<{p}-security-context>>
```

File: docs/orchestrating-elastic-stack-applications/elasticsearch/security-context.asciidoc [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  nodeSets:
+  - name: default
+    count: 3
+    podTemplate:
+      spec:
+        securityContext:
+          runAsUser: 1234 <1>
+          fsGroup: 1234 <2>
+----
+
```
Comment (pebrc) [inline docs/orchestrating-elastic-stack-applications/elasticsearch/security-context.asciidoc:41]: I think you are not supposed to have a newline here

File: docs/orchestrating-elastic-stack-applications/elasticsearch/security-context.asciidoc [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+== Run as non root Elasticsearch
+
+By default, the Elastisearch container is run as root and it is its entrypoint that is responsible to run the Elasticsearch process with the `elasticsearch` user (defined with ID 1000). In the background, ECK makes sure for you via an `initContainer` that the data volume is writeable for the `elasticsearch` user.
+
+If you choose to run the Elastisearch container as a non-root user, then you need to configure the Elasticsearch manifest with an appropriate security context to make the data volume writeable to the `elasticsearch` user by specifying the right group ID through the `fsGroup`.
+
+Kubernetes recursively changes ownership and permissions for the contents of each volume to match the `fsGroup` specified in a Pod's securityContext when that volume is mounted and makes all processes of the containers part of the supplementary group ID.
+
+For example, if you force the Pod to run as user `1234`, you need to define the `fsGroup` accordingly to `1234`:
```
Comment (pebrc) [inline docs/orchestrating-elastic-stack-applications/elasticsearch/security-context.asciidoc:21]:
```suggestion
For example, if you force the Pod to run as user `1234`, you need to set `fsGroup` accordingly to `1234`:
```

**Review by pebrc — APPROVED** [review]
LGTM

File: docs/orchestrating-elastic-stack-applications/elasticsearch/security-context.asciidoc [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+[id="{p}-{page_id}"]
+= Security Context
+
+In Kubernetes, a https://kubernetes.io/docs/tasks/configure-pod-container/security-context/[`securityContext`] defines privilege and access control settings for a Pod or Container. You can set up it through the `podTemplate` section of an Elastic resource specification.
+
+== Run as non-root Elasticsearch
+
+By default, the Elastisearch container is run as root and it is its entrypoint that is responsible to run the Elasticsearch process with the `elasticsearch` user (defined with ID 1000). In the background, ECK makes sure for you via an `initContainer` that the data volume is writeable for the `elasticsearch` user.
+
+If you choose to run the Elastisearch container as a non-root user, then you need to configure the Elasticsearch manifest with an appropriate security context to make the data volume writeable to the `elasticsearch` user by specifying the right group ID through the `fsGroup`.
```
Comment (alaudazzi) [inline docs/orchestrating-elastic-stack-applications/elasticsearch/security-context.asciidoc:17]:
```suggestion
To run the Elastisearch container as a non-root user, you need to configure the Elasticsearch manifest with an appropriate security context to make the data volume writable to the `elasticsearch` user by specifying the right group ID through the `fsGroup`.
```

**Review by alaudazzi — APPROVED** [review]
A few minor edits, otherwise LGTM.

**Pull request merged**